internal/datacoord/garbage_collector.go

```diff
@@ -236,7 +238,9 @@ func (gc *garbageCollector) clearEtcd() {
 
 	for _, segment := range drops {
 		log := log.With(zap.Int64("segmentID", segment.ID))
-		if !gc.isExpire(segment.GetDroppedAt()) {
+		to, isCompacted := compactTo[segment.GetID()]
+		// for compacted segment, try to clean up the files as long as target segment is there
+		if !isCompacted && !gc.isExpire(segment.GetDroppedAt()) {
```

Comment:
In which case will a segment be dropped but have no compacted new segment?

Reply:
> 

when the collection is dropped